api/v1alpha2/conversion_test.go

```diff
@@ -17,15 +17,39 @@ limitations under the License.
 package v1alpha2
 
 import (
+	"math/rand"
 	"testing"
 
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/diff"
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/cluster-api/api/v1alpha3"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 )
 
+func TestFuzzyConversion(t *testing.T) {
+	g := NewWithT(t)
+
+	scheme := runtime.NewScheme()
+	g.Expect(AddToScheme(scheme)).To(Succeed())
+	g.Expect(v1alpha3.AddToScheme(scheme)).To(Succeed())
+	fuzzer := utilconversion.GetFuzzer(scheme)
+
+	for i := 0; i < rand.Intn(100); i++ {
```

Comment:
You could probably convert this into a helper function that takes in `interface{}`s for src, dst, so in the api packages you'd just have to set up a scheme and call the helper func with the pairings you want to test (v1a2.Machine, v1a3.Machine; v1a2.Cluster, v1a3.Cluster; ...)

Reply:
ptal